File: app/models/ExamType.java
// SPDX-FileCopyrightText: 2024 The members of the EXAM Consortium
//
// SPDX-License-Identifier: EUPL-1.2

package models;

import jakarta.persistence.Entity;
import models.base.GeneratedIdentityModel;

@Entity
public class ExamType extends GeneratedIdentityModel {

    private String type;

    private boolean deprecated;

    public ExamType(String type) {
        this.type = type;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public void setDeprecated(boolean deprecated) {
        this.deprecated = deprecated;
    }

    public boolean isDeprecated() {
        return deprecated;
    }
}

File: app/models/exam/ExamType.scala
// SPDX-FileCopyrightText: 2024 The members of the EXAM Consortium
//
// SPDX-License-Identifier: EUPL-1.2

package models.exam

import jakarta.persistence.Entity
import models.base.GeneratedIdentityModel

import scala.compiletime.uninitialized

@Entity
class ExamType extends GeneratedIdentityModel:
  var `type`: String      = uninitialized
  var deprecated: Boolean = false
